plugins/base/src/main/kotlin/translators/descriptors/ExternalDocumentablesProvider.kt

```diff
@@ -0,0 +1,10 @@
+package org.jetbrains.dokka.base.translators.descriptors
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.DClasslike
+
+
+interface ExternalDocumentablesProvider {
+    fun findClasslike(dri: DRI, sourceSet: DokkaConfiguration.DokkaSourceSet): DClasslike?
+}
```

Comment:
I really liked the explanation for this extension point in the PR description, can you please leave a similar KDoc here or somplace else?  Preferably with an example/usecase (about supertypes from external libraries or something)? 

So that new plugin developers/maintainers can understand when to use it and why it exists, thanks!